backend/src/routers/user_auth/utils.py

```diff
@@ -51,11 +56,14 @@ def get_password_hash(password):
     return PWD_CONTEXT.hash(password)
 
 
-def get_user(db, username: str):
-    if username in db:
-        user_dict = db[username]
-        return UserInDB(**user_dict)
-    return None
+def get_user(username: str):
+    try:
+        user = User.objects.get(username=username)
+    except DoesNotExist:
+        return None
+    except MultipleObjectsReturned:
+        logging.error("%s - critical error in database, multiple instances of user.")
```

Comment:
Is this even possible to have multiple users with the same data?
If so, then this is an issue of your implementation.